### PR TITLE
Add a technote for the compiler driver functionality

### PR DESCRIPTION
**Pull request by riftEmber**
Add a technical note about the experimental compiler driver mode, containing a summary, motivation, description of control flags, and future work.

Follow up to https://github.com/chapel-lang/chapel/pull/21063 which added the driver mode.

[reviewer info placeholder]

Testing:
- [x] locally generated doc looks correct

### CODE REVIEW

File: doc/rst/technotes/driver.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   well-tested for correctness or performance.
+   If you use it and encounter a bug or limitation not yet documented as a
+   `Github issue <https://github.com/chapel-lang/chapel/issues>`_, consider
+   filing an issue as described in :ref:`readme-bugs`.
+
+The Chapel compiler ``chpl`` currently runs as a single executable responsible
+for compilation, assembly, and linking, with some components done in
+subprocesses depending on which backend is used. By contrast, many compiler
+executables (including ``gcc`` and ``clang``) are actually compiler drivers, a
+smaller driver program which is responsible for processing arguments and
```
Comment (DanilaFe) [inline doc/rst/technotes/driver.rst:19]: "compiler driver**s**" is plural, but the subsequent definition ("a smaller driver program") is singular. How about:

> By contrast, many compiler executables (including `gcc` and `clang`) are actually _compiler drivers_. A compiler driver is a smaller...

File: doc/rst/technotes/driver.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+invoking separate processes for the different components of compilation
+required.
```
Comment (DanilaFe) [inline doc/rst/technotes/driver.rst:21]: components => stages?

File: doc/rst/technotes/driver.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+With release 1.32, the Chapel compiler provides an opt-in experimental compiler
+driver mode that can be used via the ``--compiler-driver`` flag; at some point
+in the future this will become the default mode of operation. The driver
```
Comment (DanilaFe) [inline doc/rst/technotes/driver.rst:25]: This is a lot of words in a row. How about:

> With release 1.32, the Chapel compiler provides an opt-in compiler driver mode that can be used via the ``--compiler-driver`` flag. This mode is opt-in, but will become the default at some point in the future.

**Review by DanilaFe — APPROVED** [review]

File: doc/rst/technotes/driver.rst (hunk 3 above)
Comment (riftEmber) [reply doc/rst/technotes/driver.rst:25]: I like that except for the repetition of "opt-in", so I'll leave out the second instance of it.

**Pull request merged**